### PR TITLE
updating samples. changed constants class to use package constructor

### PR DESCRIPTION
**Pull request by sundararajana**
fixed to use package private constructors for constants classes. piggybacking to
include latest samples.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * Jorn Vernee ([jvernee](@JornVernee) - Committer) ⚠️ Review applies to 69ce50de0e84310fff526d53e22bc5c72634485d

### Download
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/292/head:pull/292`
`$ git checkout pull/292`

### CODE REVIEW

File: doc/panama_jextract.md [hunk 1]
```diff
@@ -754,3 +754,137 @@ java -XstartOnFirstThread -Dforeign.restricted=permit --add-modules jdk.incubato
...
+
+    public static void main(String[] args) {
+        if (args.length == 0) {
+            System.err.println("java ASTPrinter <C source or header>");
+            System.exit(1);
+        }
+
+        // parse the C header/source passed from the command line
+        var index = clang_createIndex(0, 0);
+        var tu = clang_parseTranslationUnit(index, toCString(args[0]),
```
Comment (JornVernee) [inline doc/panama_jextract.md:842]: This `toCString` should be allocated in the native scope below, so it gets cleaned up as well.

File: doc/panama_jextract.md [hunk 2]
```diff
@@ -754,3 +754,137 @@ java -XstartOnFirstThread -Dforeign.restricted=permit --add-modules jdk.incubato
...
+            NULL, 0, NULL, 0, CXTranslationUnit_None());
+
+        try (var scope = NativeScope.unboundedScope()) {
+            // array trick to update within lambda
+            var level = new int[1];
+            var visitor = new MemorySegment[1];
```
Comment (JornVernee) [inline doc/panama_jextract.md:848]: FWIW, I think you could avoid both of these by using an anonymous class instead of a lambda

**Review by JornVernee — APPROVED** [review]

File: doc/panama_jextract.md (hunk 1 above)
Comment (sundararajana) [reply doc/panama_jextract.md:842]: agree

File: doc/panama_jextract.md (hunk 2 above)
Comment (sundararajana) [reply doc/panama_jextract.md:848]: While we can take care of level param that way, the function pointer address (visitor) is used inside to call recursively. Unfortunately we need array hack for that.

**Pull request closed without merging**